app/src/test/java/com/keylesspalace/tusky/SFragmentTest.kt

```diff
@@ -0,0 +1,301 @@
+
+    private val status = Status(
+            "1",
+            statusQuery,
+            account,
+            null,
+            null,
+            null,
+            SpannedString("omgwat"),
+            Date(),
+            Collections.emptyList(),
+            0,
+            0,
+            false,
+            false,
+            false,
+            "",
+            Status.Visibility.PUBLIC,
+            arrayOf(),
+            arrayOf(),
+            null
+    )
+    private val statusCallback = FakeSearchResults(status)
+
+    @Before
+    fun setup() {
+        fragment = FakeSFragment()
+
+        apiMock = Mockito.mock(MastodonApi::class.java)
+        `when`(apiMock.search(eq(accountQuery), ArgumentMatchers.anyBoolean())).thenReturn(accountCallback)
+        `when`(apiMock.search(eq(statusQuery), ArgumentMatchers.anyBoolean())).thenReturn(statusCallback)
+        `when`(apiMock.search(eq(nonMastodonQuery), ArgumentMatchers.anyBoolean())).thenReturn(emptyCallback)
+        fragment.mastodonApi = apiMock
+    }
+
+    @Test
+    fun matchesMastodonUrls()
+    {
+        val urls = listOf(
+                "https://mastodon.foo.bar/@User",
+                "http://mastodon.foo.bar/@abc123",
+                "https://mastodon.foo.bar/@user/345667890345678",
+                "https://mastodon.foo.bar/@user/3",
+                "https://pleroma.foo.bar/users/meh3223",
+                "https://pleroma.foo.bar/users/2345",
+                "https://pleroma.foo.bar/notice/9",
+                "https://pleroma.foo.bar/notice/9345678",
+                "https://pleroma.foo.bar/objects/abcdef-123-abcd-9876543"
+        )
+
+        for(url in urls)
+            Assert.assertTrue("Failed to match $url", SFragment.looksLikeMastodonUrl(url));
+    }
+
+    @Test
+    fun rejectsNonMastodonUrls()
+    {
+        val urls = listOf(
+                "https://google.com/",
+                "https://mastodon.foo.bar/@User?foo=bar",
+                "https://mastodon.foo.bar/@User#foo",
+                "http://mastodon.foo.bar/@",
+                "http://mastodon.foo.bar/@/345678",
+                "https://mastodon.foo.bar/@user/345667890345678/",
+                "https://mastodon.foo.bar/@user/3abce",
+                "https://pleroma.foo.bar/users/",
+                "https://pleroma.foo.bar/user/2345",
+                "https://pleroma.foo.bar/notice/wat",
+                "https://pleroma.foo.bar/notices/123456",
+                "https://pleroma.foo.bar/object/abcdef-123-abcd-9876543",
+                "https://pleroma.foo.bar/objects/xabcdef-123-abcd-9876543",
+                "https://pleroma.foo.bar/objects/xabcdef-123-abcd-9876543/",
+                "https://pleroma.foo.bar/objects/xabcdef-123-abcd_9876543"
+        )
+
+        for(url in urls)
+            Assert.assertFalse("Matched invalid url $url", SFragment.looksLikeMastodonUrl(url));
+    }
+
+    @Test
+    fun beginEndSearch_setIsSearching() {
```

Comment:
All your tests have multiple `assert`s. It may make it unclear which test case fails. I'd rather see them as separate tests in each case. This way we're also sure that one test doesn't affect another one (which is kinda really important)

Reply:
Just to be clear, you'd be ok with me making these parameterized?
I.e. the url tests would all be a single parameterized test class with one test per url, and the search ones would be the same tests run on account, status, and non-mastodon urls.